src/main/java/org/wso2/carbon/inbound/iso8583/listening/ConnectionRequestHandler.java

```diff
@@ -59,10 +65,38 @@ public ConnectionRequestHandler(Socket connection, InboundProcessorParams params
      */
     public void connect() throws IOException {
         if (connection.isConnected()) {
-            String fromClient = inputStreamReader.readUTF();
-            outToClient.writeBytes("ISOMessage from " + Thread.currentThread().getName() + " is consumed :");
-            ISOMsg isoMessage = unpackRequest(fromClient);
-            msgInject.inject(isoMessage);
+            try {
+                String fromClient = "";
+                if (headerLength == 2 || headerLength == 4) {
```

Comment:
can you add a comment on the logic here? why header length is compared for the mentioned values

Reply:
Added